cachi2/core/package_managers/rpm/main.py

```diff
@@ -31,6 +33,122 @@
+        log.debug("RPM package attributes for '%s': %s", rpm_filepath, package)
+        return package
+
+    @staticmethod
+    def _query_rpm_fields(file_path: Path) -> dict[str, Optional[str]]:
+        """Query a set of RPM tags.
+
+        Tags which are optional and not set won't be returned in the resulting dict.
+        """
+        ret = {}
+        query_format = (
+            # all nvra macros should be present/mandatory in RPM
+            "name=%{NAME}\n"
+            "version=%{VERSION}\n"
+            "release=%{RELEASE}\n"
+            "arch=%{ARCH}\n"
+            # vendor and epoch are optional RPM tags; return "" if not set instead of "(None)"
+            "vendor=%|VENDOR?{%{VENDOR}}:{}|\n"
+            "epoch=%|EPOCH?{%{EPOCH}}:{}|\n"
+        )
+        rpm_args = [
+            "-q",
+            "--queryformat",
+            query_format.strip(),
+            str(file_path),
+        ]
+        rpm_fields = run_cmd(cmd=["rpm", *rpm_args], params={})
+        for entry in rpm_fields.split("\n"):
+            key, value = entry.partition("=")[::2]
+            if not value:
+                continue
+            ret[key] = value if value else None
+
+        return ret
+
+    @property
+    def purl(self) -> str:
+        """Get the purl for this package."""
+        # TODO: get rid of these mappings the moment the upstream PURL spec provides clear
+        # guidelines where does the namespace value come from, i.e. not the VENDOR RPM header tag
+        vendor_namespace_mapping = {
+            "Red Hat": "redhat",  # common Vendor string 'Red Hat Inc.'
+            "Fedora": "fedora",  # common Vendor string: Fedora Project, Fedora Copr - group @XYZ
+            "SUSE": "suse",  # common Vendor string: SUSE LLC <https://www.suse.com/>
+        }
+
+        qualifier_fields = {
+            "epoch": self.epoch,
+            "arch": self.arch,
+            "repository_id": self.repository_id,
+            "checksum": self.checksum,
+            "download_url": None if self.repository_id else quote(self.download_url),
+        }
+
+        # VENDOR tag is optional (under Informative package tags) [1]
+        # [1] https://rpm-software-management.github.io/rpm/manual/tags.html
+        if not self.vendor:
+            namespace = ""
+        else:
+            for mapping, namespc in vendor_namespace_mapping.items():
+                if mapping in self.vendor:
+                    namespace = namespc
+                    break
+            else:
+                # vendor string not recognized, normalize it in a very basic, best effort manner
+                namespace = self.vendor.lower().replace(" ", "_")
+
+        namespace = "/" + namespace if namespace else ""
+        qualifiers = "&".join([f"{k}={v}" for k, v in qualifier_fields.items() if v])
+        return f"pkg:rpm{namespace}/{self.name}@{self.version}-{self.release}?{qualifiers}"
+
+    def to_component(self, lockfile_path: Path) -> Component:
```

Comment:
Theoretically, but not right now, because we only support a single lockfile path (which is pretty static). The same situation would arise with other package managers if multiple lockfiles were supported, so if anything, it's expected I think.

Reply:
> Theoretically, but not right now, because we only support a single lockfile path (which is pretty static). The same situation would arise with other package managers if multiple lockfiles were supported, so if anything, it's expected I think.

Correction! This can happen if the user repo hosts multiple "packages" (or RPM namespaces if you will) each hosting its own lockfile. After all, the latest integration test update reflects this exact situation leading to potentially duplicate entries in the SBOM. Admittedly, SBOM entry duplication (even without my changes in this PR) is an issue that should be resolved, but given that multi-namespace RPM repo isn't something that is going to be commonly applied dedup is outside of scope of this PR.